lib/plugins/aws/info/index.js

```diff
@@ -1,154 +1,132 @@
 
 const BbPromise = require('bluebird');
+const validate = require('../lib/validate');
 const SDK = require('../');
+const chalk = require('chalk');
 
 class AwsInfo {
   constructor(serverless, options) {
     this.serverless = serverless;
     this.options = options || {};
     this.provider = 'aws';
     this.sdk = new SDK(serverless);
-
-    this.options.stage = this.options.stage
-      || (this.serverless.service.defaults && this.serverless.service.defaults.stage)
-      || 'dev';
-    this.options.region = this.options.region
-      || (this.serverless.service.defaults && this.serverless.service.defaults.region)
-      || 'us-east-1';
+    Object.assign(this, validate);
 
     this.hooks = {
       'info:info': () => BbPromise.bind(this)
           .then(this.validate)
-          .then(this.info),
+          .then(this.gather)
+          .then(this.display),
 
       'deploy:deploy': () => BbPromise.bind(this)
           .then(this.validate)
-          .then(this.info),
+          .then(this.gather)
+          .then(this.display),
     };
   }
 
-  validate() {
-    if (!this.serverless.config.servicePath) {
-      throw new this.serverless.classes.Error('This command can only be run inside a service.');
-    }
+  /**
+   * Gather information about the service
+   */
+  gather() {
+    const stackName = this.sdk.getStackName(this.options.stage);
+    const info = {
+      service: this.serverless.service.service,
+      stage: this.options.stage,
+      region: this.options.region,
+      accountId: '',
+    };
 
-    // validate stage/region/function exists in service
-    this.serverless.service.getStage(this.options.stage);
-    this.serverless.service.getRegionInStage(this.options.stage, this.options.region);
+		// Get info from CLF Outputs
+
+    return this.sdk.request('CloudFormation',
+      'describeStacks',
+      { StackName: stackName },
+      this.options.stage,
+      this.options.region)
+    .then((result) => {
+      if (result) {
+        const outputs = result.Stacks[0].Outputs;
+
+        // Account id
+        info.accountId = this.sdk.getAccountId(
+          outputs.find((x) => (x.OutputKey === 'IamRoleArnLambda')).OutputValue);
+
+        // Functions
+        info.functions = [];
+        outputs.filter(x => x.OutputKey.match(/^Function\d+Arn$/))
+          .forEach(x => {
+            const functionInfo = {};
+            functionInfo.arn = x.OutputValue;
+            functionInfo.name = functionInfo.arn.substring(x.OutputValue.lastIndexOf(':') + 1);
+            info.functions.push(functionInfo);
+          });
+
+        // Endpoints
+        info.endpoints = [];
+        outputs.filter(x => x.OutputKey.match(/^Endpoint\d+$/))
+          .forEach(x => {
+            const endpointInfo = { endpoint: x.OutputValue };
+            info.endpoints.push(endpointInfo);
+          });
+
+        // Resources
+        info.resources = [];
+      }
+
+      return BbPromise.resolve(info);
+    })
+    .catch((e) => {
```

Comment:
@bennybauer what was the issue? @eahefnawy @pmuens you're the sinon experts, any Idea?